src/dex-ui/pages/dao/MintNFTModal.tsx

```diff
@@ -0,0 +1,107 @@
+import { useState } from "react";
+import { Flex, SimpleGrid, Text } from "@chakra-ui/react";
+import { AlertDialog, Button, Color, FormInputList } from "@dex-ui-components";
+import { useFieldArray, useForm } from "react-hook-form";
+import { MintNFTTokensFormData } from "./types";
+import { MirrorNodeTokenById } from "@services";
+
+export interface MintNFTModalProps {
+  token: MirrorNodeTokenById;
+  handleMintNFT?: (tokenLinks: string[]) => void;
+}
+export const MintNFTModal = (props: MintNFTModalProps) => {
+  const { token, handleMintNFT } = props;
+  const [dialogsOpenState, setDialogsOpenState] = useState(false);
+  const defaultFieldValue = { value: "" };
+  const maxSupply = Number(token.data.max_supply);
+  const totalSupply = Number(token.data.total_supply);
+  let fieldButtonText = "+ Add Link";
+
+  const mintNFTTokenForm = useForm<MintNFTTokensFormData>({
+    defaultValues: {
+      tokenLinks: [defaultFieldValue],
+    },
+  });
+
+  const { control, register, getValues } = mintNFTTokenForm;
+
+  const { fields, append, remove } = useFieldArray({
+    control,
+    name: "tokenLinks",
+    rules: { maxLength: 4 },
+  });
+
+  if (totalSupply + getValues().tokenLinks.length >= maxSupply) {
+    fieldButtonText = "";
+  }
```

Comment:
we can simplify this `  const fieldButtonText = totalSupply + getValues().tokenLinks.length >= maxSupply ? "" : "+ Add Link";`

Reply:
You are right. made the suggested changes.